src/pages/ScholarshipForm.jsx

```diff
@@ -0,0 +1,202 @@
+import React, { useState } from 'react';
+import {
+  Container,
+  TextField,
+  Button,
+  makeStyles,
+  Select,
+  MenuItem,
+  FormControl,
+  InputLabel,
+  Checkbox,
+  FormControlLabel,
+} from '@material-ui/core';
+import 'date-fns';
+import DateFnsUtils from '@date-io/date-fns';
+import {
+  MuiPickersUtilsProvider,
+  KeyboardDatePicker,
+} from '@material-ui/pickers';
+import { ScholarshipFields } from './formFields';
+
+const useStyles = makeStyles({
+  fieldStyle: {
+    margin: '10px 0;',
+  },
+  formControlStyle: {
+    width: '40%',
+  },
+});
+
+function ScholarshipForm() {
+  const classes = useStyles();
+
+  const [formFieldStates, setFormFieldStates] = useState({
+    name: '',
+    deadline: new Date(),
+    description: '',
+    amountType: '',
+    amount: 0,
+    minAmount: 0,
+    maxAmount: 0,
+    website: '',
+    scholarshipType: '',
+    active: false,
+  });
+
+  function handleFieldChange(index, value) {
+    setFormFieldStates((prev) => ({ ...prev, [index]: value }));
+  }
+  function submitScholarship() {
+    console.log(formFieldStates); // eslint-disable-line no-console
+  }
+
+  // displays the fields for amount depending on the type of amount
+  function displayAmountFields(index) {
+    const amountTypeState = formFieldStates.amountType;
+    if (
+      !amountTypeState ||
+      amountTypeState === 'unknown' ||
+      amountTypeState === 'fullride'
+    )
+      return <div key="none" />;
+    if (amountTypeState === 'fixed') {
+      const rewardAmountRef = ScholarshipFields.rewardAmount.amount;
+      return (
+        <TextField
+          key={rewardAmountRef.key}
+          id="amount"
+          label={rewardAmountRef.label}
+          className={classes.fieldStyle}
+          type="number"
+          InputLabelProps={{ shrink: true }}
+          fullWidth
+          InputProps={{ inputProps: { min: 0, step: 50 } }}
+          value={amountTypeState}
+          onChange={(e) =>
+            handleFieldChange(index, parseInt(e.target.value, 10) || '')
+          }
+        />
+      );
+    }
+    return ['minAmount', 'maxAmount'].map((field) => (
+      <TextField
+        key={ScholarshipFields.rewardAmount[field].key}
+        id={field}
+        label={ScholarshipFields.rewardAmount[field].label}
+        className={classes.fieldStyle}
+        type="number"
+        InputLabelProps={{ shrink: true }}
+        fullWidth
+        InputProps={{ inputProps: { min: 0, step: 50 } }}
+        value={formFieldStates[field]}
+        onChange={(e) =>
+          handleFieldChange(field, parseInt(e.target.value, 10) || '')
+        }
+      />
+    ));
+  }
+
+  function displayField(fieldKey) {
+    const { key } = ScholarshipFields[fieldKey];
+    const { label } = ScholarshipFields[fieldKey];
+    const { type } = ScholarshipFields[fieldKey];
+
+    switch (type) {
```

Comment:
Rather than having a `displayField` with a switch statement I recommend having the [elements](https://en.reactjs.org/docs/rendering-elements.html) themselves be the value in the `formFields` mapping and creating Function components for any common ones. E.g. 
```jsx
function TextField(props) {
  // Returns a <TextField> according to the passed in props
}

const formFields = {
  name: <TextField your custom props/>, // Alternatively TextField({ custom props })
  date: <MuiPickersUtilsProvider utils={DateFnsUtils} ...>
           <KeyboardDatePicker  .../>
        </MuiPickersUtilsProvider>,
  ...
};
```

Reply:
I didn't think of that but it looks cleaner! How would I pass in State and onChange handler to the elements though?